ml_api/server.py

```diff
@@ -43,13 +43,29 @@ def get_p():
             img = cv2.imdecode(img_array, -1)
             detections = detect(net_main, img, thresh=THRESH)
             return jsonify({'detections': detections})
-        except:
+        except Exception as err:
             sentry_sdk.capture_exception()
+            app.logger.error(f"Failed to get image {request.args} - {err}")
+            abort(
+                make_response(
+                    jsonify(
+                        detections=[],
+                        message=f"Failed to get image {request.args} - {err}",
+                    ),
+                    503,
+                )
+            )
     else:
-        app.logger.warn("Invalid request params: {}".format(request.args))
+        app.logger.warn(f"Invalid request params: {request.args}")
+        abort(
+            make_response(
+                jsonify(
+                    detections=[], message=f"Invalid request params: {request.args}"
+                ),
+                400,
```

Comment:
I'll be fine with 400. But the convention is probably 422. Again not that it matters that much.

Reply:
updated error code to 422.